applications/vanilla/modules/class.promotedcontentmodule.php

```diff
@@ -134,6 +142,7 @@ public function validate($parameters = []) {
      */
     public function getData() {
         $this->setData('Content', false);
+        $this->setData('ShowIfNoResults', $this->ShowIfNoResults);
```

Comment:
You shouldn't need this. This is already a property on the object. You don't need to set it in the data array.